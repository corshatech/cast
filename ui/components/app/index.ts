export * from "./Layout/Layout";
export * from "./FeatureComponent/FeatureComponent";
