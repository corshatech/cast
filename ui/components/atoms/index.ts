export * from './Button/Button';
export * from './Typography/Typography';
