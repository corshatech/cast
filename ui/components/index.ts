export * from './atoms';
export * from './app';
