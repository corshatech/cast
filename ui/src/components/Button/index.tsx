import React from "react";
import css from "./Button.module.scss";

interface ButtonProps {
  /**
   * Is this the principal call to action on the page?
   */
  primary?: boolean;
  /**
   * What background color to use
   */
  backgroundColor?: string;
  /**
   * How large should the button be?
   */
  size?: "small" | "medium" | "large";
  /**
   * Button contents
   */
  label: string;
  /**
   * Optional click handler
   */
  onClick?: () => void;
}

/**
 * Primary UI component for user interaction
 */
export const Button = ({
  primary = false,
  size = "medium",
  backgroundColor,
  label,
  ...props
}: ButtonProps) => {
  const mode = primary
    ? css["storybook-button-primary"]
    : css["storybook-button-secondary"];
  return (
    <button
      type="button"
      className={[
        css["storybook-button"],
        css[`storybook-button-${size}`],
        mode,
      ].join(" ")}
      style={{ backgroundColor }}
      {...props}
    >
      {label}
    </button>
  );
};
