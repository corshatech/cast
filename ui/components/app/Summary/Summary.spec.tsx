import renderer from 'react-test-renderer';
import { Summary, SummaryLoading } from '@/components/index';

describe('Analyses Summary Component', () => {
  it('renders no problems variant', () => {
    const tree = renderer.create(
      <Summary
        faults={0}
        scansPassed={17}
        findings={0}
        severityCounts={{ none: 0, low: 0, medium: 0, high: 0, critical: 0 }}
      />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders some problems variant', () => {
    const tree = renderer.create(
      <Summary
        faults={7}
        scansPassed={10}
        findings={143}
        severityCounts={{ none: 0, low: 2, medium: 2, high: 0, critical: 3 }}
      />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders all problems variant', () => {
    const tree = renderer.create(
      <Summary
        faults={7}
        scansPassed={10}
        findings={143}
        severityCounts={{ none: 1, low: 2, medium: 3, high: 4, critical: 5 }}
      />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Analyses Summary Loading Component', () => {
  it('rednders correctly', () => {
    const tree = renderer.create(
      <SummaryLoading />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
