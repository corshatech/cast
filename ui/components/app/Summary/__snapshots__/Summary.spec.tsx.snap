// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analyses Summary Component renders all problems variant 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <h3
      className="leading-6 text-h4 font-poppins font-semibold"
    >
      Ongoing Results
    </h3>
    <p
      className="leading-6 text-body1 font-nunito"
    >
      7 Faults (143 Findings)
    </p>
    <p
      className="leading-6 text-body1 font-nunito"
      style={
        {
          "color": "rgba(0, 0, 0, 0.6)",
        }
      }
    >
      10
       passed
    </p>
    <ul>
      <li>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
          data-testid="CrisisAlertIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14.5 2.5c0 1.5-1.5 6-1.5 6h-2S9.5 4 9.5 2.5C9.5 1.12 10.62 0 12 0s2.5 1.12 2.5 2.5zM12 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm4.08-4.89c.18-.75.33-1.47.39-2.06C19.75 4.69 22 8.08 22 12c0 5.52-4.48 10-10 10S2 17.52 2 12c0-3.92 2.25-7.31 5.53-8.95.07.59.21 1.32.39 2.06C5.58 6.51 4 9.07 4 12c0 4.42 3.58 8 8 8s8-3.58 8-8c0-2.93-1.58-5.49-3.92-6.89zM18 12c0 3.31-2.69 6-6 6s-6-2.69-6-6c0-2 .98-3.77 2.48-4.86.23.81.65 2.07.65 2.07C8.43 9.93 8 10.92 8 12c0 2.21 1.79 4 4 4s4-1.79 4-4c0-1.08-.43-2.07-1.13-2.79 0 0 .41-1.22.65-2.07C17.02 8.23 18 10 18 12z"
          />
        </svg>
         
        5
         Critical
      </li>
      <li>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
          data-testid="ErrorIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
          />
        </svg>
         
        4
         High
      </li>
      <li>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorWarning MuiSvgIcon-fontSizeMedium css-3adcp6-MuiSvgIcon-root"
          data-testid="WarningIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
         
        3
         Medium
      </li>
      <li>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorWarning MuiSvgIcon-fontSizeMedium css-3adcp6-MuiSvgIcon-root"
          data-testid="DarkModeIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9 9-4.03 9-9c0-.46-.04-.92-.1-1.36-.98 1.37-2.58 2.26-4.4 2.26-2.98 0-5.4-2.42-5.4-5.4 0-1.81.89-3.42 2.26-4.4-.44-.06-.9-.1-1.36-.1z"
          />
        </svg>
         
        2
         Low
      </li>
      <li>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-1yq1soz-MuiSvgIcon-root"
          data-testid="LightbulbIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 21c0 .5.4 1 1 1h4c.6 0 1-.5 1-1v-1H9v1zm3-19C8.1 2 5 5.1 5 9c0 2.4 1.2 4.5 3 5.7V17c0 .5.4 1 1 1h6c.6 0 1-.5 1-1v-2.3c1.8-1.3 3-3.4 3-5.7 0-3.9-3.1-7-7-7z"
          />
        </svg>
         
        1
         None
      </li>
    </ul>
  </div>
</div>
`;

exports[`Analyses Summary Component renders no problems variant 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <h3
      className="leading-6 text-h4 font-poppins font-semibold"
    >
      Ongoing Results
    </h3>
    <p
      className="leading-6 text-body1 font-nunito"
    >
      No problems detected yet. Live scanning will continue in the background.
    </p>
    <p
      className="leading-6 text-body1 font-nunito"
      style={
        {
          "color": "rgba(0, 0, 0, 0.6)",
        }
      }
    >
      17
       passed
    </p>
    <ul />
  </div>
</div>
`;

exports[`Analyses Summary Component renders some problems variant 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <h3
      className="leading-6 text-h4 font-poppins font-semibold"
    >
      Ongoing Results
    </h3>
    <p
      className="leading-6 text-body1 font-nunito"
    >
      7 Faults (143 Findings)
    </p>
    <p
      className="leading-6 text-body1 font-nunito"
      style={
        {
          "color": "rgba(0, 0, 0, 0.6)",
        }
      }
    >
      10
       passed
    </p>
    <ul>
      <li>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
          data-testid="CrisisAlertIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14.5 2.5c0 1.5-1.5 6-1.5 6h-2S9.5 4 9.5 2.5C9.5 1.12 10.62 0 12 0s2.5 1.12 2.5 2.5zM12 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm4.08-4.89c.18-.75.33-1.47.39-2.06C19.75 4.69 22 8.08 22 12c0 5.52-4.48 10-10 10S2 17.52 2 12c0-3.92 2.25-7.31 5.53-8.95.07.59.21 1.32.39 2.06C5.58 6.51 4 9.07 4 12c0 4.42 3.58 8 8 8s8-3.58 8-8c0-2.93-1.58-5.49-3.92-6.89zM18 12c0 3.31-2.69 6-6 6s-6-2.69-6-6c0-2 .98-3.77 2.48-4.86.23.81.65 2.07.65 2.07C8.43 9.93 8 10.92 8 12c0 2.21 1.79 4 4 4s4-1.79 4-4c0-1.08-.43-2.07-1.13-2.79 0 0 .41-1.22.65-2.07C17.02 8.23 18 10 18 12z"
          />
        </svg>
         
        3
         Critical
      </li>
      <li>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorWarning MuiSvgIcon-fontSizeMedium css-3adcp6-MuiSvgIcon-root"
          data-testid="WarningIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
         
        2
         Medium
      </li>
      <li>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorWarning MuiSvgIcon-fontSizeMedium css-3adcp6-MuiSvgIcon-root"
          data-testid="DarkModeIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9 9-4.03 9-9c0-.46-.04-.92-.1-1.36-.98 1.37-2.58 2.26-4.4 2.26-2.98 0-5.4-2.42-5.4-5.4 0-1.81.89-3.42 2.26-4.4-.44-.06-.9-.1-1.36-.1z"
          />
        </svg>
         
        2
         Low
      </li>
    </ul>
  </div>
</div>
`;
